src/snowflake/connector/s3_storage_client.py

```diff
@@ -94,7 +94,9 @@ def transfer_accelerate_config(
         self, use_accelerate_endpoint: bool | None = None
     ) -> bool:
         # accelerate cannot be used in China and us government
-        if self.location_type and "S3China" in self.location_type:
+        if (self.location_type and "S3China" in self.location_type) or (
+            self.region_name and "cn" in self.region_name
```

Comment:
1. add test
2. double check the region that cn doesn't get used in other areas to avoid accidentally disable acc point for non-china users

Reply:
test is added

Reply:
I confirmed with @sfc-gh-schen that this is a valid fix that will not accidentally turn off acceleration for other regions